src/components/Modal.jsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { Modal as WP2Modal, Button, Icon } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { store as editPostStore } from '@wordpress/edit-post';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { launch } from './Icons';
+import { store as nfdComingSoonStore } from '../store';
+import { setComingSoon } from '../utils/api/comingSoon';
+
+
+const Modal = () => {
+	const { setIsModalOpen } = useDispatch(nfdComingSoonStore);
+	const [isComingSoonActive, setIsComingSoonActive] = useState(true); 
+
+	const { isModalOpen } = useSelect(
+		(select) => ({
+			isModalOpen: select(nfdComingSoonStore).isModalOpen(),
+		})
+	);
+
+	const handlePublishAndLaunch = () => {
+		setComingSoon(false);
+		setIsComingSoonActive(false);
+		setIsModalOpen(false);
+	};
+
+	if (!isModalOpen || !isComingSoonActive) {
+		return null;
+	}
+
+	const title = __('Ready to launch your Site?', 'nfd-coming-soon');
+	const heading = __('Pages you publish will not be visible to the public until you launch your site.', 'nfd-coming-soon');
+	const launchButtonText = __('Publish & Launch Site', 'nfd-coming-soon');
+	const withoutLaunchButtonText = __('Publish without launching', 'nfd-coming-soon');
```

Comment:
I assume we should change this to say "Cancel" or "Not Now" or something along those lines.

Reply:
I think Cancel sounds too much like "don't do either thing". We know that at the time that this modal is shown the post is already published on the WordPress side and the only thing the modal really controls is the coming soon page, but it might not be obvious to users. Let's go with the copy in the ticket for now. This seems to match that, when I compare it to the screenshot, so I think we're good. https://jira.newfold.com/secure/attachment/1938664/f3d7c7bc7de3d764eecea6ae8dbcd2eb.png